tools/ctexplain/bazel_api.py

```diff
@@ -0,0 +1,161 @@
+# Lint as: python3
+# Copyright 2020 The Bazel Authors. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""API for Bazel calls for config, cquery, and required fragment info.
+
+There's no Python Bazel API so we invoke Bazel as a subprocess.
+"""
+import json
+import os
+import re
+import subprocess
+from tools.ctexplain.types import Configuration
+from tools.ctexplain.types import ConfiguredTarget
+from tools.ctexplain.types import HostConfiguration
+from tools.ctexplain.types import NullConfiguration
+
+
+def run_bazel_in_client(args):
+  """Calls bazel within the current workspace.
+
+  For production use. Tests use an alternative invoker that goes through test
+  infrastructure.
+
+  Args:
+    args: the arguments to call Bazel with
+
+  Returns:
+    Tuple of (return code, stdout, stderr)
+  """
+  result = subprocess.run(
+      ["bazel"] + args,
+      cwd=os.getcwd(),
+      stdout=subprocess.PIPE,
+      stderr=subprocess.PIPE,
+      check=True)
+  return [
+      result.returncode,
+      result.stdout.decode("utf-8").split(os.linesep), result.stderr
+  ]
+
+
+class BazelApi():
+  """API that accepts injectable Bazel invocation logic."""
+
+  def __init__(self, run_bazel=run_bazel_in_client):
```

Comment:
Since no pytype, have a docstring here clarifying the type of run_bazel
(looks like a `Callable[[List[str]], Tuple[int, ???, ???]]` although feel free to format as more readable like `List[str] -> (int, ???, ???)`)

??? is whatever type stdout and stderr actually are above (str? List[str]?)